main/modules/settings/installed-versions/services/ides-installation-status.service.ts

```diff
@@ -16,15 +17,23 @@ export default class IDEsInstallationStatus {
   async getDevonfwIDEsStatus(
     onNotify: (instances: IdeVersions) => void
   ): Promise<void> {
-    const versions = await this.versions.getInstalledVersions();
+    const localIDEs = await this.localIDEs.getInstalledDevonfwIDEs();
     const ides = await this.idesRetriever.getDevonfwIDEs();
 
+    for (const localIDE of localIDEs) {
+      onNotify({
+        ...localIDE,
+        updated: formatDate(
+          ides.find((ide) => ide.version === localIDE.version).updated
+        ),
+        changelog: await this.changelogService.hasChangelog(localIDE.version),
+      });
+    }
+
     for (const ide of ides) {
       onNotify({
         ...ide,
-        installed: versions.includes(ide.version),
         changelog: await this.changelogService.hasChangelog(ide.version),
-        downloading: false,
       });
```

Comment:
Will the content of `localIDEs ` be a subset of `ides`? Because `localIDEs` are the same as `ides` but with the restriction of "being installed". In this situation you will display duplicated values on the table. 
You should remove the `localIDEs` from `ides`, something like:

```ts
const localIDEs = await this.localIDEs.getInstalledDevonfwIDEs();
const ides = await this.idesRetriever.getDevonfwIDEs();
const notInstalledIDEs = ides.filter(IsNotIncludedInLocal)

for(cont localIDE of localIDEs) { ... }
for(cont ide of notInstalledIDEs ) { ... }
```

Also I have some doubs about the sorting. Won't you be displaying first the local and then the non installed in the table? Before it was sorted by version number

Reply:
There could be a case where a user has multiple working copies of the same devonfw distribution (for whatever reasons). For example I could have the 2020.04.001 version in 2 different locations in my system with different projects in them. In that case, `localIDEs` being a subset of `ides` would ignore such copies and display any version only once in the list. 
There could be another case where a user might be having, say 2020.04.004 version already, but wants to have another copy of this same version. In such case, the **Download** button should be visible to download a fresh copy along with the **Uninstall** button for the installed copy. `localIDEs` being a subset of `ides` wont allow that (since **Download** and **Uninstall** buttons are displayed in the same column in the table).
For the sorting, I'm purposely displaying the local versions first since the page is called **Installed Versions**